src/test/java/com/metamx/common/io/smoosh/SmooshedFileMapperTest.java

```diff
@@ -71,6 +71,142 @@ public void testSanity() throws Exception
+	    	  File tmpFile = File.createTempFile(String.format("smoosh-%s", i), ".bin");
+	          Files.write(Ints.toByteArray(i), tmpFile);
+	          smoosher.add(String.format("%d", i), tmpFile);
+	          tmpFile.delete();
+	      }      
+	      smoosher.close();
+	      
+	      File[] files = baseDir.listFiles();
+	      Arrays.sort(files);
+
+	      Assert.assertEquals(5, files.length); // 4 smoosh files and 1 meta file
+	      for (int i = 0; i < 4; ++i) {
+	        Assert.assertEquals(FileSmoosher.makeChunkFile(baseDir, i), files[i]);
+	      }
+	      Assert.assertEquals(FileSmoosher.metaFile(baseDir), files[files.length - 1]);
+
+	      SmooshedFileMapper mapper = SmooshedFileMapper.load(baseDir);
+	      for (int i = 0; i < 20; ++i) {
+	        ByteBuffer buf = mapper.mapFile(String.format("%d", i));
+	        Assert.assertEquals(0, buf.position());
+	        Assert.assertEquals(4, buf.remaining());
+	        Assert.assertEquals(4, buf.capacity());
+	        Assert.assertEquals(i, buf.getInt());
+	      }
+	      mapper.close();
+	    }
+	    finally {
+	      for (File file : baseDir.listFiles()) {
+	        file.delete();
+	      }
+	    }
+  }
+  
+  @Test
+  public void testForMultipleSmooshedFileMapperOpenedAndClosed() throws Exception
+  {
+	    File baseDir = Files.createTempDir();
+
+	    try {
+	      FileSmoosher smoosher = new FileSmoosher(baseDir, 21);
+	      
+	      final SmooshedWriter writer = smoosher.addWithSmooshedWriter(String.format("%d", 19), 4);
+          
+	      for (int i = 0; i < 19; ++i) {
+	    	  File tmpFile = File.createTempFile(String.format("smoosh-%s", i), ".bin");
+	          Files.write(Ints.toByteArray(i), tmpFile);
+	          smoosher.add(String.format("%d", i), tmpFile);
+	          if(i==10)
+	          {
+	    	      writer.write(ByteBuffer.wrap(Ints.toByteArray(19)));
+	    	      CloseQuietly.close(writer);
+	          }
+	          tmpFile.delete();
+	      }    
+	      smoosher.close();
+	      
+	      File[] files = baseDir.listFiles();
+	      Arrays.sort(files);
+
+	      Assert.assertEquals(5, files.length); // 4 smoosh files and 1 meta file
+	      for (int i = 0; i < 4; ++i) {
+	        Assert.assertEquals(FileSmoosher.makeChunkFile(baseDir, i), files[i]);
+	      }
+	      Assert.assertEquals(FileSmoosher.metaFile(baseDir), files[files.length - 1]);
+
+	      SmooshedFileMapper mapper = SmooshedFileMapper.load(baseDir);
+	      for (int i = 0; i < 20; ++i) {
+	        ByteBuffer buf = mapper.mapFile(String.format("%d", i));
+	        Assert.assertEquals(0, buf.position());
+	        Assert.assertEquals(4, buf.remaining());
+	        Assert.assertEquals(4, buf.capacity());
+	        Assert.assertEquals(i, buf.getInt());
+	      }
+	      mapper.close();
+	    }
+	    finally {
+	      for (File file : baseDir.listFiles()) {
+	        file.delete();
+	      }
+	    }
+  }
+  
+  @Test
+  public void testForMultipleSmooshedFileMapperOpenedAndClosedInTheLast() throws Exception
```

Comment:
There's a bunch of code being C&P'd around here to verify the exact same thing.  Please extract the common parts and put them in a shared method.